### PR TITLE
KSE-178: Normalize usage of 'record', 'message' and 'event' in Kafka doc

### PR DESCRIPTION
**Pull request by abbccdda**
### Description 
We want to normalize the usage of the term in Kafka records, by replacing all the old usage like `message` or `event` with `record`.
Here we search for all the key word appearances in `*.md` files and do the replacement based on the context.

### Testing done 
N/A

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

**Review by spena — APPROVED** [review]
lgtm

File: docs/developer-guide/joins/synthetic-keys.md [hunk 1]
```diff
@@ -18,15 +18,15 @@ The previous statement seems straightforward enough: create a new table that's t
 performing a full outer join of two source tables, joining on their ID columns. But in a 
 full-outer join, either `L.ID` or `R.ID` may be missing (`NULL`), or both 
 may have the same value. Since the data produced to {{ site.aktm }} should always have a non-null 
-message key, ksql selects the first non-null key to use:
+record key, ksql selects the first non-null key to use:
 
-| L.ID  | R.ID | Kafka message key |
+| L.ID  | R.ID | Kafka record key |
```
Comment (JimGalasyn) [inline docs/developer-guide/joins/synthetic-keys.md:23]:
```suggestion
| L.ID  | R.ID | Kafka record key  |
```

File: docs/developer-guide/ksqldb-reference/create-stream.md [hunk 2]
```diff
@@ -31,7 +31,7 @@ key. This situation is handled differently by [ksqlDB TABLEs](../create-table),
 |                          |  STREAM                                                       | TABLE                                                             |
 | ------------------------ | --------------------------------------------------------------| ----------------------------------------------------------------- |
 | Key column type          | `KEY`                                                         | `PRIMARY KEY`                                                     |
-| NON NULL key constraint  | No                                                            | Yes <br> Messages in the Kafka topic with a NULL `PRIMARY KEY` are ignored. |
+| NON NULL key constraint  | No                                                            | Yes <br> Records in the Kafka topic with a NULL `PRIMARY KEY` are ignored. |
```
Comment (JimGalasyn) [inline docs/developer-guide/ksqldb-reference/create-stream.md:34]:
```suggestion
| NON NULL key constraint  | No                                                            | Yes <br> Records in the Kafka topic with a NULL `PRIMARY KEY` are ignored.  |

```

File: docs/developer-guide/ksqldb-reference/create-stream.md [hunk 3]
```diff
@@ -74,7 +74,7 @@ The WITH clause supports the following properties:
...
 | VALUE_DELIMITER         | Used when VALUE_FORMAT='DELIMITED'. Supports single character to be a delimiter, defaults to ','. For space and tab delimited values you must use the special values 'SPACE' or 'TAB', not an actual space or tab character. |
-| TIMESTAMP               | By default, the pseudo `ROWTIME` column is the timestamp of the message in the Kafka topic. The TIMESTAMP property can be used to override `ROWTIME` with the contents of the specified column within the Kafka message (similar to timestamp extractors in Kafka's Streams API). Timestamps have a millisecond accuracy. Time-based operations, such as windowing, will process a record according to the timestamp in `ROWTIME`. |
+| TIMESTAMP               | By default, the pseudo `ROWTIME` column is the timestamp of the record in the Kafka topic. The TIMESTAMP property can be used to override `ROWTIME` with the contents of the specified column within the Kafka message (similar to timestamp extractors in Kafka's Streams API). Timestamps have a millisecond accuracy. Time-based operations, such as windowing, will process a record according to the timestamp in `ROWTIME`. |
```
Comment (JimGalasyn) [inline docs/developer-guide/ksqldb-reference/create-stream.md:77]:
```suggestion
| TIMESTAMP               | By default, the pseudo `ROWTIME` column is the timestamp of the record in the Kafka topic. The TIMESTAMP property can be used to override `ROWTIME` with the contents of the specified column within the Kafka message (similar to timestamp extractors in Kafka's Streams API). Timestamps have a millisecond accuracy. Time-based operations, such as windowing, will process a record according to the timestamp in `ROWTIME`.  |

```

**Review by JimGalasyn — APPROVED** [review]
LGTM, thanks for taking this on, @abbccdda!

**Review by JimGalasyn — APPROVED** [review]
LGTM!

**Pull request merged**